das/worker.go

```diff
@@ -71,12 +76,23 @@ func (w *worker) run(
 		}
 		w.setResult(curr, err)
 		metrics.observeSample(ctx, h, time.Since(startSample), err)
+
 		if err != nil {
 			log.Debugw("failed to sampled header", "height", h.Height(), "hash", h.Hash(),
 				"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "err", err)
-		} else {
-			log.Debugw("sampled header", "height", h.Height(), "hash", h.Hash(),
-				"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "finished (s)", time.Since(startSample))
+			continue
+		}
+
+		log.Debugw("sampled header", "height", h.Height(), "hash", h.Hash(),
+			"square width", len(h.DAH.RowsRoots), "data root", h.DAH.Hash(), "finished (s)", time.Since(startSample))
+
+		// notify network about recent header being available to be synced via shrex.Getter
```

Comment:
```suggestion
		// notify network via shrexsub that we are available as a peer for syncing the recent data
		// light nodes provide a stub `shrexsub.BroadcastFn`, as they do not run shrexsub 
```
@renaynay and others please help find a good comment, don't think mine is great either

Reply:
```suggestion
		// notify network about availability of new block data (note: only full nodes can notify)
```